### PR TITLE
Add integration test to check that catalogNumber is specified

### PR DESCRIPTION
**Pull request by mukeshk**
Addressed to #1065
Validation test on series search.
Checks for empty value in search field.

### CODE REVIEW

File: src/test/robotframework/series/search/validation.robot [hunk 1]
```diff
@@ -0,0 +1,15 @@
+*** Settings ***
+Documentation    Verify series search validation scenarios
+Library          SeleniumLibrary
+Suite Setup      Before Test Suite
+Suite Teardown   Close Browser
+Force Tags       series  search  validation  fireme
+
+*** Test Cases ***
+Catalog Number is required for series search
```
Comment (php-coder) [inline src/test/robotframework/series/search/validation.robot]: Could you, please, rename the test case to "Search the series with empty required field".

File: src/test/robotframework/series/search/validation.robot [hunk 2]
```diff
@@ -0,0 +1,15 @@
+*** Settings ***
+Documentation    Verify series search validation scenarios
+Library          SeleniumLibrary
+Suite Setup      Before Test Suite
+Suite Teardown   Close Browser
+Force Tags       series  search  validation  fireme
```
[the remark this answers is not in the document]
Comment (mukeshk) [reply src/test/robotframework/series/search/validation.robot]: Yes. will do

File: src/test/robotframework/series/search/validation.robot (hunk 1 above)
Comment (mukeshk) [reply src/test/robotframework/series/search/validation.robot]: will do

**Comment by mukeshk** [comment]
Do I need to create a new pull? I have made the changes which you have suggested and used git commit - command option for committing the changes.  
What would be the next step?

**Comment by php-coder** [comment]
> Do I need to create a new pull? I have made the changes which you have suggested and used git commit - command option for committing the changes.

No, all the feedback to this PR should be part of this PR. So, after editing the files and add new changes, you commit them and push. All PRs are connected to the related git branches. When you add/modify commits in the branch, these changes become available in the PR.

**Comment by mukeshk** [comment]
OK. Thanks. Teris is reporting build failed. On my instance, I ran the
robot test suite it worked.
Not sure if i broke something

**Comment by php-coder** [comment]
@mukeshk Travis CI is broken because Danger detects that you have 5 commits instead of 1. See its comment: https://github.com/php-coder/mystamps/pull/1077#issuecomment-510770321

When you fix that, it should become green.

File: src/test/robotframework/series/search/validation.robot [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+*** Test Cases ***
+Catalog Number is required for series search
+	Submit Form             id=search-series-form
+	Element Text Should Be  id=catalogNumber.errors  Value must not be empty
+
+*** Keywords ***
+Before Test Suite
+	Open Browser                        ${SITE_URL}/  ${BROWSER}
```
[the remark this answers is not in the document]
Comment (mukeshk) [reply src/test/robotframework/series/search/validation.robot]: Added the keyword.

**Comment by php-coder** [comment]
@mukeshk Could you also update your author name, so it won't contain dot and quotes?

```console
$ git config user.name "Mukesh Katariya"
$ git commit --am --reset-author
```

**Comment by php-coder** [comment]
@mukeshk The last one (I hope): in this repo all commit messages must follow Conventional Commits convention (see https://www.conventionalcommits.org). Could you, please, amend the commit to looks like this:

```
test(series/search/validation.robot): add test to check that catalog number is specified.
```

Thanks!

**Comment by mukeshk** [comment]
> @mukeshk The last one (I hope): in this repo all commit messages must follow Conventional Commits convention (see https://www.conventionalcommits.org). Could you, please, amend the commit to looks like this:
> 
> ```
> test(series/search/validation.robot): add test to check that catalog number is specified.
> ```
> 
> Thanks!
should the comment be like this, 
                                 
"test(series/search/validation.robot): add test to check that catalog number is specified.                                                 
Addressed to #1065"

**Comment by mukeshk** [comment]
Well, I am not able to figure out. I added the comment as suggested.
Not sure again travis fails.

**Comment by php-coder** [comment]
>  Not sure again travis fails.

Here is what Danger bot suggests:

> danger check: pull request contains merge commits! Please, rebase your branch to get rid of them: git rebase master gh1065_search_series_by_catalog and git push --force-with-lease

Have you tried that? It seems that you have because now there less commits, so, try to run it again.

Also, why you have these merge commits at all?

**Comment by php-coder** [comment]
@mukeshk I see that you made a commit to my `master` branch. It's good that you found that typo and fixed it but, please, don't push to my repository :) Propose the changes in PRs instead. Thank you!

**Comment by mukeshk** [comment]
> git rebase master gh1065_search_series_by_catalog
First, rewinding head to replay your work on top of it...
Applying: Implemeted Issues 1065
Applying: test(series/search/validation.robot): add test to check that
catalog number is specified.
Using index info to reconstruct a base tree...
M       src/main/webapp/WEB-INF/views/site/index.html
Falling back to patching base and 3-way merge...
No changes -- Patch already applied.

> git push --force-with-lease
fatal: HttpRequestException encountered.
   An error occurred while sending the request.
Username for 'https://github.com': jmukesh@gmail.com
Password for 'https://jmukesh@gmail.com@github.com':
Counting objects: 14, done.
Delta compression using up to 4 threads.
Compressing objects: 100% (13/13), done.
Writing objects: 100% (14/14), 1.39 KiB | 1.39 MiB/s, done.
Total 14 (delta 7), reused 0 (delta 0)
remote: Resolving deltas: 100% (7/7), completed with 7 local objects.
To https://github.com/mukeshk/mystamps.git
 + 2419e1ce...3d0c99bb gh1065_search_series_by_catalog ->
gh1065_search_series_by_catalog (forced update)

Is this good ?

**Comment by php-coder** [comment]
@mukeshk It's better now, thank you! But I don't see that my comments were addressed.

**Comment by mukeshk** [comment]
Ok. Sure will follow.

**Comment by mukeshk** [comment]
OK.
I did the fireme - removal.
Added the keyword.
Rename the test case name.

Only last part the comment to the commit.  which i am somehow losing out
when i do the command you suggested.

**Comment by mukeshk** [comment]
Is it good now?  my changes got lost across the commits and Travis fixes I was trying out.

**Comment by php-coder** [comment]
Yeah, I see your changes now and one more squash is required.

If you don't want to make many squashes, you can use `git commit --am -a next` time, it will add new changes to the last commit.

**Comment by mukeshk** [comment]
Yeah, I see your changes now and one more squash is required.
- what should I do here?

If you don't want to make many squashes, you can use git commit --am
-a next time,
it will add new changes to the last commit.
 - I will study about git command.

**Comment by php-coder** [comment]
> what should I do here?

See the bot's message: https://github.com/php-coder/mystamps/pull/1077/commits#issuecomment-510770321

There is a link to article about how to do squash. If it doesn't answer on how to do that, ask me then :)

**Comment by mukeshk** [comment]
Thanks for your reply.
I will check through it and let you know

**Comment by php-coder** [comment]
> @mukeshk Could you also update your author name, so it won't contain dot and quotes?

Don't forget to update the name after squash.

**Comment by php-coder** [comment]
> Don't forget to update the name after squash.

@mukeshk The only one thing is left ^^^

**Comment by php-coder** [comment]
@mukeshk Yes, you did it!

The last: could you edit the commit message to remove unrelated sentences?
At this point it looks like this:
```
test(series/search/validation.robot): add test to check that catalog number is specified.

Address to #1065

Update Validation robot

Update
```
and it should be
```
test(series/search/validation.robot): add test to check that catalog number is specified.

Fix #1065
```

**Comment by mukeshk** [comment]
Apologies for that.
Thanks. I learnt a lot today.
Thanks for your help and quick response.

Best regards
Mukesh

**Comment by php-coder** [comment]
@mukeshk Thanks again for the PR! I've added you to the list of contributors: https://github.com/php-coder/mystamps/wiki/team-members

If you want to cleanup, here are the commands: https://github.com/php-coder/mystamps/wiki/after_merge_steps#remove-merged-branch

The next step, is to sync your master branch with mine and create a new branch on top of updated master for the next task. Let me know if you need any help with that.

**Comment by mukeshk** [comment]
Thanks, for all the help through this.
I will check for submitting the next issue.

Thanks again for your help.

**Comment by mukeshk** [comment]
Followed the steps you suggested.
Submitted new PR.

**Pull request merged**